### ISSUE TITLE
Nuki 1.0 stopped working since 6.2+

### ISSUE THREAD

**Issue by leroyloren** [technyon/nuki_hub#60]
After upgrading to version 6.2+, communication with Nuki 1.0 stopped working, pairing will not happen after upgrading.
Re-pairing won't help too.

When downgrading to 6.1, everything works immediately.

![obrazek](https://user-images.githubusercontent.com/57643470/208389070-6b9d4d2e-158e-4f28-857b-62d1bb9fccc1.png)

**Comment by todde99** [comment]
The same was happening with my Nuki 1. I switch to "register as app" and the pairing was working again.

**Comment by technyon** [comment]
@leroyloren Does it work for you is you set register as app?

**Comment by leroyloren** [comment]
Tried "set register as app" (including completed (re)pairing instructions), issue is the same.

**Comment by technyon** [comment]
I have updated two libraries, let's try reverting them one by one. First the NUKI library ... can you check if it works with the attached firmware? When flashing is successful it the version in the web interface should read "6.6-rvt-nuki-ble".

[nuki_hub-rvt-nuki-ble.zip](https://github.com/technyon/nuki_hub/files/10260960/nuki_hub-rvt-nuki-ble.zip)

**Comment by leroyloren** [comment]
With `nuki_hub-rvt-nuki-ble.zip`

It works fine.
![obrazek](https://user-images.githubusercontent.com/57643470/208484550-2189a546-3aa6-440b-81e6-53fc4cc94d70.png)

**Comment by technyon** [comment]
ok thanks, I'll get in touch with the maintainers of that library.

**Comment by Mincka** [comment]
I came on this issue because I did not manage to pair my Smart Lock 3.0 with 6.6
The issue was not fixed with `6.6-rvt-nuki-ble`. I went back to 6.1 and the pairing worked immediately.
Then OTA update to 6.6 and the pairing worked but the autodiscovery publishing failed, as described in #61.
I don't need to open another issue since I managed to find a way to pair it but others with 3.0 lock may also get stuck with 6.6.

**Comment by technyon** [comment]
Ok we'll have to try one thing at a time. I've removed a line of code that was added that drastically shortens the bluetooth connection timeout. Please give it a try and let me know.

[nuki_hub-tst-1.zip](https://github.com/technyon/nuki_hub/files/10280200/nuki_hub-tst-1.zip)

**Comment by Mincka** [comment]
Still not working.
```
MQTT Connected | Yes
NUKI Lock paired | Yes
NUKI Lock state | undefined
Firmware | 6.6-tst-1

```

**Comment by technyon** [comment]
Please try the attached firmware, only the bluetooth library is reverted in this one.

If this doesn't fix it, I'll prepare several build going thorough relevant commits of the nuki ble library and see which one breakes thing.

[nuki_hub-rvt-nimble.zip](https://github.com/technyon/nuki_hub/files/10296602/nuki_hub-rvt-nimble.zip)

**Comment by technyon** [comment]
Please test the attached firmware files ... unfortunately I can't reproduce the problem in lack of a NUKI v1. In each binary I've reverted the nuki ble lib to a previous version. 1 is the oldest, 4 the latest and actually quite close to the released version. I hope we nail it down to one specific commit like this.

[nuki_hub-rvt-nuki-ble-1-0565965.zip](https://github.com/technyon/nuki_hub/files/10297064/nuki_hub-rvt-nuki-ble-1-0565965.zip)
[nuki_hub-rvt-nuki-ble-2-832aa3d.zip](https://github.com/technyon/nuki_hub/files/10297065/nuki_hub-rvt-nuki-ble-2-832aa3d.zip)
[nuki_hub-rvt-nuki-ble-3-0f9b233.zip](https://github.com/technyon/nuki_hub/files/10297066/nuki_hub-rvt-nuki-ble-3-0f9b233.zip)
[nuki_hub-rvt-nuki-ble-4-9499a0c.zip](https://github.com/technyon/nuki_hub/files/10297067/nuki_hub-rvt-nuki-ble-4-9499a0c.zip)

**Comment by Mincka** [comment]
Thanks a lot for your support. ❤️ I tested all the nuki ble versions and none of them worked.

However, the rvt nimble worked immediately after the firmware upgrade. 🎉
![image](https://user-images.githubusercontent.com/6184289/209427622-7d4ea4f4-da68-4a27-bd17-17a4099d0e02.png)

In my case, I don't have a Nuki v1 but a refurbished **Nuki Smart Lock 3.0** (Non Pro).
Firmware 3.4.10

It's very strange because, like I said, I have another 3.0 purchased 6 months ago and no issues with the 6.6 release.
Maybe there are different hardware versions of the 3.0 and for some reason, one of them is not working with the latest version of nimble.

**Comment by technyon** [comment]
It seems I'll downgrade the nimble library then.

@leroyloren Could you give it a try too.  I think the original issue with NUKI v1 is unrelated to this, since downgrading the nuki library fixed it for you. It would be really helpful to know which change breaks things for v1

**Comment by leroyloren** [comment]
Ok, I'll try and let you know the result

**Comment by leroyloren** [comment]
![obrazek](https://user-images.githubusercontent.com/57643470/209429397-7c0d28d2-d29c-427e-accd-88468e789a45.png)
---
![obrazek](https://user-images.githubusercontent.com/57643470/209429084-da0a7baa-e13b-428f-8a3c-1b7652a1d5f0.png)
---
![obrazek](https://user-images.githubusercontent.com/57643470/209429314-7387ce87-5274-4cad-83ba-562ae7b235ab.png)
subjectively slower respond
---
![obrazek](https://user-images.githubusercontent.com/57643470/209430267-49c376aa-dfc1-4567-a256-9c973a473eac.png)
responds with a long delay when locking manually (maybe just a feeling?)
---
![obrazek](https://user-images.githubusercontent.com/57643470/209430312-11ceb475-6628-4923-a26c-fe7b4c8d9c9f.png)
states no change
---

### Resume:
ble-1 works best

**Comment by Mincka** [comment]
I wanted to try a last time the official 6.6 just to be sure on my side. The Lock state stayed "undefined" and for some other unknown reasons, the OTA update feature never wanted to come back, it stayed in "please wait" state. I had to flash it with Espressif tools to revert to 6.1. Now, I did the OTA update with "6.6-rvt-nimble" again and I can confirm it works just fine for one of my Smart Lock 3.0.

I don't know if this is something possible but it would be great to be able to either pick a release or upload a specific nuki_hub.bin in the web flasher. I can open a separate issue for this as a feature request if it makes sense.

**Comment by mundschenk-at** [comment]
I have noticed a similar issue when my Opener was having a funk (and I thought it had to do with the FW version). I had to disable it because OTA part never got initialized otherwise.

**Comment by Mincka** [comment]
6.7 is able to communicate with the lock, however, I see buggy messages pushed to MQTT (as seen in #51)

For instance, this topic is recreated each time the ESP32 boots:
![image](https://user-images.githubusercontent.com/6184289/209436997-ea8f588a-5438-4ef6-9c96-da88bcafe739.png)
The nukiopener check box is uncheked.

I also add a failure once (as seen in #62) but not sure why.
![image](https://user-images.githubusercontent.com/6184289/209437310-e92757cd-864d-4ad4-a2a9-d3f83e4e6f92.png)

And I also saw this message in the logs once:
`Lock action received: a1`

Maybe the failed lock actions are related to the buggy messages.

**Comment by technyon** [comment]
@Mincka Could you try one more binary? I found an issue that could be related and tried the suggested fix:

https://github.com/h2zero/NimBLE-Arduino/issues/473

Maybe we can keep nimble 1.4.1 after all.

[nuki_hub-nimble-1.4.1-tryfix.zip](https://github.com/technyon/nuki_hub/files/10298829/nuki_hub-nimble-1.4.1-tryfix.zip)

**Comment by Mincka** [comment]
Congrats, it works. 👍 
![image](https://user-images.githubusercontent.com/6184289/209444691-91ef1a0d-e22b-4a81-8d8b-de74b6a6529a.png)
It's not a blocking issue but the `nukiopener` topic is still created on boot.

**Comment by technyon** [comment]
Thanks ... I'll keep that for the next release since it doesn't really make a big difference. Makes you wonder why the change the default of that parameter so that it's not within bluetooth specs.

**Comment by leroyloren** [comment]
This version doesn't work for me with Nuki 1.0

![obrazek](https://user-images.githubusercontent.com/57643470/209451137-c3065fe5-10a7-4cc3-a91f-11b0e43d375c.png)

![obrazek](https://user-images.githubusercontent.com/57643470/209451144-ae6c3be6-eca0-4b94-a754-dc45c90ee2d1.png)

**Comment by Mincka** [comment]
@leroyloren I am not using 6.7 but this version [nuki_hub-nimble-1.4.1-tryfix.zip](https://github.com/technyon/nuki_hub/files/10298829/nuki_hub-nimble-1.4.1-tryfix.zip) which have the latest version of the lib but used with different parameters. Can you try this one instead?

**Comment by technyon** [comment]
@leroyloren It's getting a bit confusing with all those binaries. To make it clear, when you have time please test the binaries that have a downgraded Nuki library version:

[nuki_hub-rvt-nuki-ble-1-0565965.zip](https://github.com/technyon/nuki_hub/files/10297064/nuki_hub-rvt-nuki-ble-1-0565965.zip)
[nuki_hub-rvt-nuki-ble-2-832aa3d.zip](https://github.com/technyon/nuki_hub/files/10297065/nuki_hub-rvt-nuki-ble-2-832aa3d.zip)
[nuki_hub-rvt-nuki-ble-3-0f9b233.zip](https://github.com/technyon/nuki_hub/files/10297066/nuki_hub-rvt-nuki-ble-3-0f9b233.zip)
[nuki_hub-rvt-nuki-ble-4-9499a0c.zip](https://github.com/technyon/nuki_hub/files/10297067/nuki_hub-rvt-nuki-ble-4-9499a0c.zip)

I consider the issue with the Nuki v3 seperate, and fixed now.

**Comment by leroyloren** [comment]
here https://github.com/technyon/nuki_hub/issues/60#issuecomment-1364499861 I wrote the behavior of all versions, with the fact that `nuki_hub-rvt-nuki-ble-1-0565965` worked best in my case, from which I subsequently understood that you reverted the NimBLE library in v6.7, but after the upgrade the issue repeats itself.

p.s. can you please clarify the difference between "register as app / bridge" it's not very clear from the description and I don't know how much it can be related to my issue.

**Comment by technyon** [comment]
ok sorry I'm losing track of all the comments :D. But basically everything up to "nuki-ble-4-9499a0c" does work? I don't see undefined state with -4. What do mean by "states no change".

Register as app / bridge is just a flag when pairing the device. If registering as a bridge, it'll unpair any bridge that's already paired to the device to prevent two bridges managing the lock. If selecting register as app, you can pair as many devices including ESPs as you like. Once paired, there shouldn't be a difference in behaviour.

**Comment by technyon** [comment]
P.S.:

I found one more suspicious commit. Can you try this binary?

[nuki_hub-7.6-nuki-ble-5-0f9b233.zip](https://github.com/technyon/nuki_hub/files/10299785/nuki_hub-7.6-nuki-ble-5-0f9b233.zip)

**Comment by leroyloren** [comment]
I tried this version. After upload seems everything is ok, but reactions (lock/unlock) was are special. For example. I had to 3-4 times click for lock, 2 times for unlock etc. After about an hour nuki stopped respond for commands. After restart hub, everything seems ok, but nuki is not respond for commands.

_nuki-ble-4-9499a0c - states no change_  State has changed only one time and nuki stopped the respond for commands

**Comment by technyon** [comment]
Great the commit that touches all files breaks things. Let's first make sure though it's this commit. I've reverted that specific commit in the attached binary, but kept all other changes including later commits. Can you give it another try?

[nuki_hub-6.7-revert-nuki-ble-9499a0.zip](https://github.com/technyon/nuki_hub/files/10300527/nuki_hub-6.7-revert-nuki-ble-9499a0.zip)

**Comment by leroyloren** [comment]
After update, Lock paired: No, Lock state: Undefined. 
Again I paired Nuki. Lock paired: Yes, Lock state: Undefined.
After restart hub, Nuki paired status is agains No.
Nuki is not respond for any commands.

**Comment by technyon** [comment]
I'm out of ideas, the changes I make and the behaviour don't seem to be consistent. Could you recheck the four binaries rvt-nuki-ble-1 to -4 to make sure your findings are consistent?

**Comment by todde99** [comment]
I'm also having a Nuki V1 (firmware version 1.14.1) and had the same issue like @leroyloren . 

I did some tests with the 4 different versions and also with version 6.7. The result is really wired! I have no idea why 6.6 was not paring with the Nuki V1.

Current working config:
Hostname	nukihub
MQTT Connected	Yes
NUKI Lock paired	Yes
NUKI Lock state	unlocked
register as bridge
Firmware	6.6-rvt-nuki-ble
Tests: unlatch > working; lock > working; unlock > working (action is triggered immediately)

ble-1:
Hostname	nukihub
MQTT Connected	Yes
NUKI Lock paired	Yes
NUKI Lock state	unlocked
register as bridge
Firmware	6.6-rvt-nuki-ble-1-0565965
Tests: unlatch > working; lock > working; unlock > working (action is triggered immediately)

Hostname	nukihub
MQTT Connected	Yes
NUKI Lock paired	Yes
NUKI Lock state	unlocked
register as bridge
Firmware	6.6-rvt-nuki-ble-2-832aa3d
Tests: unlatch > working; lock > working; unlock > working (action is triggered immediately)

Hostname	nukihub
MQTT Connected	Yes
NUKI Lock paired	Yes
NUKI Lock state	unlocked
register as bridge
Firmware	6.6-rvt-nuki-ble-3-0f9b233
Tests: unlatch > working; lock > working; unlock > working (action is triggered immediately)

Hostname	nukihub
MQTT Connected	Yes
NUKI Lock paired	Yes
NUKI Lock state	unlocked
register as bridge
Firmware	6.6-rvt-nuki-ble-4-9499a0c
Tests: unlatch > working; lock > working; unlock > working (takes ~10 sec until action is triggered)

and finally:
Hostname	nukihub
MQTT Connected	Yes
NUKI Lock paired	Yes
NUKI Lock state	unlocked
Firmware	6.7
Tests: unlatch > working; lock > working; unlock > working (takes ~10 sec until action is triggered)

**Comment by technyon** [comment]
Let's give it another try then. There are three statemachines in the library that are central to the bluetooth communication. They have all been changed in commit 9499a0c. I've undone the changes to the state machines from that commit and built another binary:

[nuki_hub-6.7-rvt-cmd-stm.zip](https://github.com/technyon/nuki_hub/files/10326286/nuki_hub-6.7-rvt-cmd-stm.zip)

**Comment by todde99** [comment]
For me this build is working with my Nuki V1 perfectly

Hostname	nukihub
MQTT Connected	Yes
NUKI Lock paired	Yes
NUKI Lock state	unlocked
Firmware	6.7-rvt-cmd-stm
Tests: unlatch > working; lock > working; unlock > working (action is triggered immediately)

Thank you @technyon for all your work you are doing here!

**Comment by lucaoliano** [comment]
Hi @technyon,
first of all thank you for your great work.

I have a Nuki Lock 3.0 and I also have problems with version 6.2+

This morning I tried unsuccessfully the last version 6.8, here the logs:
<img width="664" alt="1 4 1-be1fa97" src="https://user-images.githubusercontent.com/16431719/210394995-5da26e0c-18b5-4d66-9150-6f8508154378.png">

I tried downgrading the libraries NimBLE-Arduino and nuki_ble and I had success only with NimBLE-Arduino 1.4.0 and nuki_ble d63a9ff, same versions of 6.1.

Here the logs of my tests:

- NimBLE-Arduino 1.4.0 and nuki_ble be1fa97
<img width="667" alt="1 4 0-be1fa97" src="https://user-images.githubusercontent.com/16431719/210394438-e5c2ec6a-9fa2-4a03-99e7-ef66a8b547fa.png">

- NimBLE-Arduino 1.4.1 and nuki_ble d63a9ff
<img width="491" alt="1 4 1-d63a9ff" src="https://user-images.githubusercontent.com/16431719/210395243-8abe58c4-8f99-4210-b613-294bcdd668f9.png">

**Comment by technyon** [comment]
Please open a new issue, it's hard to keep track of all of this in one issue. Does this happen always or sometimes?

**Comment by lucaoliano** [comment]
I've opened a new issue.

**Comment by technyon** [comment]
The fix for v1 is in Release 6.9
